### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Horisofine**
Updated the README directly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,20 @@
-# Gym2Real_isaacgym
-This repo provides a custom task to be used with Isaac Gym as well as scripts for training and exporting models.
+# TWIP - Isaac Sim
+This repo contains the code and configuration used during our journey of developing TWIP using NVIDIA's Isaac Sim and rl games.
 
 ## Requirements
+- Ubutun 22.04 LTS (Works with pop_os as well)
```
Comment (AugustDG) [inline README.md]:
```suggestion
- Ubuntu 22.04 LTS (Works with Pop OS as well)
```

File: README.md [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-
-### IsaacGymEnvs Setup
-- `git clone https://github.com/NVIDIA-Omniverse/IsaacGymEnvs` (tested with commit `9656bac7e59b96382d2c5040b90d2ea5c227d56d`)
-- `cd IsaacGymEnvs && pip install -e .` to install Python package
+- Isaac Sim (tested with version `4.0.0`)
+- RL games
+- Anaconda | Miniconda
+
+## Isaac Sim Setup
+- Download Isaac Sim by following the steps found within Nvidia's installation guide https://docs.omniverse.nvidia.com/isaacsim/latest/installation/install_workstation.html
```
Comment (AugustDG) [inline README.md]:
```suggestion
- Download Isaac Sim by following the steps found within Nvidia's installation (guide)[https://docs.omniverse.nvidia.com/isaacsim/latest/installation/install_workstation.html].
```

File: README.md [hunk 3]
```diff
@@ -24,29 +25,34 @@ This repo provides a custom task to be used with Isaac Gym as well as scripts fo
...
-- `cfg/train/{TaskName}PPO.yaml`: Config gile defining hyperparameters for training
 
-### Training
+### Running Isaac Sim
+The entry point of our project is `twip.py`. To run our project you must configure the environment:
+- `conda activate isaac-sim`
+- `source _isaac_sim/setup_conda_env.sh`
+- `python twip.py --sim-only --num-envs 4`
+
+`--sim-only` is used to run the simluation alone without any reinforcement learning. `--num-envs` specifies the number of twip environments within the simulation. To see all the arguments, check out the source code of `twip.py`.
```
Comment (AugustDG) [inline README.md]:
```suggestion
`--sim-only` is used to run the simluation alone without any reinforcement learning. `--num-envs` specifies the number of twip environments within the simulation. To see all the options, run `twip.py --help`.
```

File: README.md [hunk 4]
```diff
@@ -24,29 +25,34 @@ This repo provides a custom task to be used with Isaac Gym as well as scripts fo
...
+
+### Training (?)
 - `python train.py task={task_name}` to begin training
 - Models are saved as `runs/{TaskName}/nn/{checkpoint_name}.pth`
 
-### Exporting ONNX
+### Exporting ONNX (?)
 - `python export.py task={task_name} checkpoint="runs/{TaskName}/nn/{checkpoint_name}.pth"` to export
```
Comment (AugustDG) [inline README.md]:
```suggestion
 - `python twip.py --checkpoint="runs/{checkpoint_name}/nn/{task_name}.pth --export-onnx"` to export
```

File: README.md [hunk 5]
```diff
@@ -24,29 +25,34 @@ This repo provides a custom task to be used with Isaac Gym as well as scripts fo
...
 
-## IsaacGymEnvs
-This repo was designed to be used after installing IsaacGymEnvs (`https://github.com/NVIDIA-Omniverse/IsaacGymEnvs`) as a Python library.
-IsaacGymEnvs comes close to being usable as a library, but requires adding your custom task to a map.
+### Adding a New Task (?)
+Create a new folder called `tasks/{task_name}`. The following files will be needed:
+- `tasks/{task_name}/{task_name}.py`: Class defining the task. Rewards, observations, actions are all defined here.
+- `tasks/{task_name}/assets/`: URDF and meshes should be placed here.
+- `cfg/task/{TaskName}.yaml`: Config file defining environment parameters and domain randomization
+- `cfg/train/{TaskName}PPO.yaml`: Config file defining hyperparameters for training
```
Comment (AugustDG) [inline README.md:50]: Are we doing this? I don't think the directory structure is currently this one.

File: README.md [hunk 6]
```diff
@@ -24,29 +25,34 @@ This repo provides a custom task to be used with Isaac Gym as well as scripts fo
...
 
 The following files are taken from IsaacGymEnvs and modified so that this isn't necessary:
  - `cfg/config.yaml` (minimal changes)
  - `utils/rlgames_utils.py` (minimal changes)
  - `base/vec_task.py` (minimal changes)
  - `train.py`
```
Comment (AugustDG) [inline README.md]: We don't have these files anymore

File: README.md [hunk 7]
```diff
@@ -24,29 +25,34 @@ This repo provides a custom task to be used with Isaac Gym as well as scripts fo
...
 
-Instead of looking for tasks in a static map, this code uses pydoc.locate to dynamically load the task class.
\ No newline at end of file
+Instead of looking for tasks in a static map, this code uses pydoc.locate to dynamically load the task class.
```
Comment (AugustDG) [inline README.md]: We don't do this either

File: README.md [hunk 8]
```diff
@@ -24,29 +25,34 @@ This repo provides a custom task to be used with Isaac Gym as well as scripts fo
...
+`--sim-only` is used to run the simluation alone without any reinforcement learning. `--num-envs` specifies the number of twip environments within the simulation. To see all the arguments, check out the source code of `twip.py`.
+
+### Training (?)
 - `python train.py task={task_name}` to begin training
 - Models are saved as `runs/{TaskName}/nn/{checkpoint_name}.pth`
 
-### Exporting ONNX
+### Exporting ONNX (?)
 - `python export.py task={task_name} checkpoint="runs/{TaskName}/nn/{checkpoint_name}.pth"` to export
 - Model is exported to `runs/{TaskName}/nn/{checkpoint_name}.pth.onnx`
```
Comment (AugustDG) [inline README.md]:
```suggestion
 - Model is exported to `runs/{checkpoint_name}/nn/{task_name}.pth.onnx`
```

File: README.md [hunk 9]
```diff
@@ -24,29 +25,34 @@ This repo provides a custom task to be used with Isaac Gym as well as scripts fo
...
+- `python twip.py --sim-only --num-envs 4`
+
+`--sim-only` is used to run the simluation alone without any reinforcement learning. `--num-envs` specifies the number of twip environments within the simulation. To see all the arguments, check out the source code of `twip.py`.
+
+### Training (?)
 - `python train.py task={task_name}` to begin training
 - Models are saved as `runs/{TaskName}/nn/{checkpoint_name}.pth`
 
-### Exporting ONNX
+### Exporting ONNX (?)
```
Comment (AugustDG) [inline README.md]:
```suggestion
### Exporting ONNX
```

File: README.md [hunk 10]
```diff
@@ -24,29 +25,34 @@ This repo provides a custom task to be used with Isaac Gym as well as scripts fo
...
-### Training
+### Running Isaac Sim
+The entry point of our project is `twip.py`. To run our project you must configure the environment:
+- `conda activate isaac-sim`
+- `source _isaac_sim/setup_conda_env.sh`
+- `python twip.py --sim-only --num-envs 4`
+
+`--sim-only` is used to run the simluation alone without any reinforcement learning. `--num-envs` specifies the number of twip environments within the simulation. To see all the arguments, check out the source code of `twip.py`.
+
+### Training (?)
```
Comment (AugustDG) [inline README.md]:
```suggestion
### Training
```

File: README.md [hunk 11]
```diff
@@ -1,19 +1,20 @@
...
+## Isaac Sim Setup
+- Download Isaac Sim by following the steps found within Nvidia's installation guide https://docs.omniverse.nvidia.com/isaacsim/latest/installation/install_workstation.html
+- Run Isaac sim from the Omniverse application to make sure it runs properly
+- Clone this repository `git clone https://github.com/TheNewtonCapstone/twip-isaac-sim`
+- Remove any folder named `_isaac_sim`
+- Run `cd twip-isaac-sim && ln -s home/YOUR_NAME/.local/share/ov/pkg/isaac-sim-4.0.0`
+- Create the conda environment `conda env create -f environment.yml` 
 
 ## Repo Structure
 - `train.py`: Main script used for training/testing models
```
Comment (AugustDG) [inline README.md]:
```suggestion
 - `twip.py`: Main script used for training/testing/exporting models
```

File: README.md [hunk 12]
```diff
@@ -1,19 +1,20 @@
-# Gym2Real_isaacgym
-This repo provides a custom task to be used with Isaac Gym as well as scripts for training and exporting models.
+# TWIP - Isaac Sim
+This repo contains the code and configuration used during our journey of developing TWIP using NVIDIA's Isaac Sim and rl games.
 
 ## Requirements
+- Ubutun 22.04 LTS (Works with pop_os as well)
 - Nvidia GPU with 470+ drivers (`nvidia-smi` to make sure these are set up)
```
Comment (AugustDG) [inline README.md]:
```suggestion
 - Nvidia GPU with 510.73.05+ drivers (`nvidia-smi` to make sure these are set up)
```

**Review by AugustDG — CHANGES_REQUESTED** [review]
Hopefully this isn't too harsh, let me know if that's the case for future reviews!

File: README.md (hunk 5 above)
Comment (Horisofine) [reply README.md:50]: I think this structure would be useful when creating different policies.

Comment (AugustDG) [reply README.md:50]: Wouldn't it just be a change of of environments? Or a flag?

Or do you prefer having different directories and we load up stuff "dynamically"?

**Pull request merged**